src/pages/index.js

```diff
@@ -1,248 +1,250 @@
+      newCardPopup.closeModal();
+    })
+    .catch((err) => {
+      console.log(err);
+    })
+    .finally(() => {
+      submitButton.textContent = originalButtonText;
+    });
+}
 
-const editProfileButton = document.querySelector(
-  "#profile__edit-button-action"
+const newUserInfo = new UserInfo(
+  "#profilename",
+  "#profilejob",
+  "#profileImage"
 );
 
+api
+  .loadUserInfo()
+  .then((userData) => {
+    newUserInfo.setUserInfo({
+      name: userData.name,
+      about: userData.about,
+      avatar: userData.avatar,
+    });
+  })
+  .catch((err) => {
+    console.log(err);
+  });
+
 editProfileButton.addEventListener("click", () => {
-  // nameInput.value = profileName.textContent;
-  // jobInput.value = profileJob.textContent;
   const data = newUserInfo.getUserInfo();
   nameInput.value = data.name;
   jobInput.value = data.job;
-  cardFormValidator.toggleButtonState();
   profileCardPopup.openModal();
 });
 
-// profileEditModalClose.addEventListener("click", () =>
-//   profileCardPopup.closeModal()
-// );
+profilePhoto.addEventListener("click", () => {
+  profilePhotoEditValidator.toggleButtonState();
+  profilePhotoEdit.openModal();
+});
 
-// modalPreviewCloseButton.addEventListener("click", () =>
-//   contentCardPreview.closeModal()
-// );
+const profileCardPopup = new PopupWithForm({
+  popupSelector: "#profileModal",
+  handleFormSubmit: handleProfileFormSubmit,
+});
 
-// ---- //Content Modal open and close event listeners -----------------------------------
+const profilePhotoEdit = new PopupWithForm({
+  popupSelector: "#profilePhotoModal",
+  handleFormSubmit: handleProfilePhotoSubmit,
+});
 
-contentAddButton.addEventListener("click", () => {
-  cardFormValidator.toggleButtonState();
-  newCardPopup.openModal();
+const deleteConfirmation = new PopupWithForm({
+  popupSelector: "#deleteContent",
+});
+
+const contentCardPreview = new PopupWithImage({
+  popupSelector: "#jsPopupSelector",
 });
-// contentAddModalClose.addEventListener("click", () => newCardPopup.closeModal());
-
-// ---- //Esc Button Modal Close ---------------------------------------------------------
-
-// function handleEscButton(event) {
-//   const key = event.key;
-//   if (key === "Escape") {
-//     const openedPopup = document.querySelector(".modal_opened");
-//     closeModal(openedPopup);
-//   }
-// }
-// //Overlay Modal Close
-// function handleOverlayClose(evt) {
-//   if (evt.target.classList.contains("modal_opened")) {
-//     closeModal(evt.target);
-//   }
-// }
-
-// const submitButtonEvent = document.querySelector(".modal__form");
-
-// submitButtonEvent.addEventListener("submit", () =>
-//   newUserInfo.setUserInfo({
-//     name: nameInput.value,
-//     about: jobInput.value,
-//   })
-// );
+
+function handleCardDeleteClick(card) {
+  const submitButton = document.querySelector("#confirmDeletion");
+  deleteConfirmation.openModal();
+  let submitButtonPressed = false;
+  submitButton.addEventListener("click", () => {
```

Comment:
Please, delete it because you add more and more the same listeners to the same button. This can cause a `memory leak`


Everything works fine without that like that


```js


function handleCardDeleteClick(card) {
  deleteConfirmation.openModal();

  deleteConfirmation.setSubmitAction(() => {
    api
      .deleteCard(card.getId())
      .then(() => {
        card.remove();
        deleteConfirmation.closeModal();
      })
      .catch((err) => {
        console.log(err);
      });
  });
}

```

Reply:
I have completed all the tasks. Regarding the second bullet point. Are you reffering to the "redundant code" or the closeModal(); functions in the .then statements of the API requests?

Reply:
I meant this  https://disk.yandex.ru/i/_7OUOgIZznWkwg